odoo/tools/mail.py

```diff
@@ -298,6 +298,24 @@ def validate_url(url):
     return url
 
 
+def get_title_from_html(html_content, encoding='utf-8'):
+    try:
+        parser = etree.HTMLParser(encoding=encoding)
+        tree = html.fromstring(html_content, parser=parser)
+        title = tree.xpath('//meta[@property="og:title"]/@content')
```

Comment:
can we keep `title = p.find('.//title').text` as a fallback if title is empty here ?

while we change it, maybe we can add a test (check that `og:title` as the priority over `<title/>`, etc)

Reply:
>  I think taking the og:title when there is one might just be better? and fallback on the <title> when there isn't?

Yes exactly :) `<title/>` is the fallback